src/renderer/browser/components/RequestsProcessor/Decrypt/index.js

```diff
@@ -0,0 +1,36 @@
+import { withCall, withData } from 'spunky';
+import { compose, withProps } from 'recompose';
+import { pick } from 'lodash';
+
+import authActions from 'login/actions/authActions';
+
+import Decrypt from './Decrypt';
+import withClean from '../../../hocs/withClean';
+import withNullLoader from '../../../hocs/withNullLoader';
+import withRejectMessage from '../../../hocs/withRejectMessage';
+
+const mapAuthDataToProps = ({ wif }) => ({ wif });
+const mapDecryptDataToProps = (data) => ({ data });
+
+const CONFIG_KEYS = ['senderPublicKey', 'iv', 'mac', 'data'];
+
+export default function makeDecrypt(decryptActions) {
+  return compose(
+    withClean(decryptActions),
+
+    withProps(({ args }) => pick(args[0], CONFIG_KEYS)),
+
+    withData(authActions, mapAuthDataToProps),
+
+    withCall(decryptActions, ({ senderPublicKey, wif, iv, mac, data }) => ({
```

Comment:
We currently use `withInitialCall` everywhere for the API, is this something we should address before merging? @mhuggins

Reply:
That would potentially be safer here to ensure we don't inadvertently call the action twice (in case of a re-render for any reason).